js/utils/serverConnect.js

```diff
@@ -325,15 +325,41 @@ export default function connect(server, options = {}) {
     if (connectAttempt) connectAttempt.cancel();
   };
 
-  // If we're not connecting to the local bundled server or it's running for a different coin,
-  // then let's ensure it's stopped.
+  if (server.get('builtIn') && !localServer) {
+    // This should never happen to normal users. The only way it would is if you are a dev
+    // and mucking with localStorage and / or fudging the source for the app to masquerade
+    // as a bundled app.
+    throw new Error('A configuration for a built-in server should only be used on ' +
+      ' the bundled app.');
+  }
+
```

Comment:
This has grown a bit more in complexity than I would have liked. In determining whether we need to bounce an already running local server, it not only needs to factor in the coin the existing server is running in, but also the zcash binary path (if zec) and also (further down) the tor settings.

It's very manual right now in that if we add any additional server start args, some code will be needed to check for them. I'd like to simplify by just comparing what start args the existing server is running with vs. what we need and if they differ, bounce. But... I need to refactor a thing or two to make that happen so I'm going to save that as a down-time task.